src/Hashing.js

```diff
@@ -1,32 +1,67 @@
 /* global require exports BROWSER_RUNTIME */
 
-const Blake2 = require('blakejs');
+const Blake2bPureJs = require('blakejs');
+const Blake2bWasm = require('blake2b-wasm');
+
 const SHA256 = require('jssha/dist/sha256');
 const SHA3 = require('jssha/dist/sha3');
 
 let lib;
 if (typeof BROWSER_RUNTIME != 'undefined' && BROWSER_RUNTIME) {
-    lib = require('@ngua/cardano-serialization-lib-browser');
+  lib = require('@ngua/cardano-serialization-lib-browser');
 } else {
-    lib = require('@ngua/cardano-serialization-lib-nodejs');
+  lib = require('@ngua/cardano-serialization-lib-nodejs');
 }
 
 const DIGEST_LENGTH_256 = 32;
 const DIGEST_LENGTH_224 = 28;
 
-exports.blake2b256Hash = bytesToHash =>
-    Blake2.blake2b(bytesToHash, null, DIGEST_LENGTH_256);
+const withBlake2bWasmObject = bytesToHash => digestLength => {
+  let obj = Blake2bWasm(digestLength);
+  obj.update(Buffer.from(bytesToHash));
+  return obj;
+};
+
+const blake2bHash = bytesToHash => digestLength => {
+  return new Promise(resolve => {
+    Blake2bWasm.ready(error => {
+      if (error || !Blake2bWasm.SUPPORTED) {
```

Comment:
Does SUPPORTED mean exactly that WASM is available? Or is the condition when SUPPORTED is `true` more strict than that?

Since CSL depends on WASM anyway, and we can't meaningfully use the library without it, we can just drop the pure JS implementation from the bundle.

Reply:
@klntsky Yes, SUPPORTED is only used to check if WASM is available, so I'll drop the pure js part

Reply:
@klntsky But on the other hand, we have to check if there is no error, so maybe it would be reasonable to have a fallback for this case?